File: generator-common/src/main/java/io/github/potjerodekool/openapi/common/StatusCodeMatcher.java
package io.github.potjerodekool.openapi.common;

import java.util.regex.Pattern;

public final class StatusCodeMatcher {

    private static final Pattern TWO_X_X = Pattern.compile("2[0-9]{2}");

    private StatusCodeMatcher() {
    }

    public static boolean is2XX(final String status) {
        return TWO_X_X.matcher(status).matches();
    }

}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/StatusCodeMatcher.kt
package io.github.potjerodekool.openapi.common

import java.util.regex.Pattern

object StatusCodeMatcher {
    private val TWO_X_X: Pattern = Pattern.compile("2[0-9]{2}")

    fun is2XX(status: String?): Boolean {
        return status != null && TWO_X_X.matcher(status).matches()
    }
}
